internal/provider/resource_access_jaas_role.go

```diff
@@ -0,0 +1,138 @@
+		ID:              roleAccess.ID,
+		Users:           roleAccess.Users,
+		Groups:          roleAccess.Groups,
+		ServiceAccounts: roleAccess.ServiceAccounts,
+		Access:          roleAccess.Access,
+	}
+	// When importing, the role name will be empty
+	var tag names.Tag
+	if roleAccess.RoleID.ValueString() != "" {
+		tag = jimmnames.NewRoleTag(roleAccess.RoleID.ValueString())
+	}
+	return accessGroup, tag
+}
+
+// Save implements the [resourceInfo] interface, used to save info on Terraform's state.
+func (j roleInfo) Save(ctx context.Context, setter Setter, info objectsWithAccess, tag names.Tag) diag.Diagnostics {
+	roleAccess := jaasAccessModelResourceRole{
+		RoleID:          basetypes.NewStringValue(tag.Id()),
+		ID:              info.ID,
+		Users:           info.Users,
+		Groups:          info.Groups,
+		ServiceAccounts: info.ServiceAccounts,
+		Access:          info.Access,
+	}
+	return setter.Set(ctx, roleAccess)
+}
+
+// ImportHint implements [resourceInfo] and provides a hint to users on the import string format.
+func (j roleInfo) ImportHint() string {
+	return "<role-uuid>:<access-level>"
+}
+
+// TagFromID validates the id to be a valid role ID
+// and returns a role tag.
+func (j roleInfo) TagFromID(id string) (names.Tag, error) {
+	if !jimmnames.IsValidRoleId(id) {
+		return nil, errors.New("invalid role ID")
+	}
+	return jimmnames.NewRoleTag(id), nil
+}
+
+type jaasAccessRoleResource struct {
+	genericJAASAccessResource
+}
+
+type jaasAccessModelResourceRole struct {
+	RoleID          types.String `tfsdk:"role_id"`
+	Users           types.Set    `tfsdk:"users"`
+	ServiceAccounts types.Set    `tfsdk:"service_accounts"`
+	Groups          types.Set    `tfsdk:"groups"`
+	Access          types.String `tfsdk:"access"`
+
+	// ID required for imports
+	ID types.String `tfsdk:"id"`
+}
+
+// Metadata returns metadata about the JAAS role access resource.
+func (a *jaasAccessRoleResource) Metadata(_ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_jaas_access_role"
+}
+
+// ConfigValidators sets validators for the resource.
+func (r *jaasAccessRoleResource) ConfigValidators(ctx context.Context) []resource.ConfigValidator {
+	return []resource.ConfigValidator{
+		NewRequiresJAASValidator(r.client),
+		resourcevalidator.AtLeastOneOf(
+			path.MatchRoot("users"),
+			path.MatchRoot("groups"),
+			path.MatchRoot("service_accounts"),
+		),
+	}
+}
+
+// Schema defines the schema for the JAAS role access resource.
+func (a *jaasAccessRoleResource) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	attributes := baseAccessSchema()
+	// we don't support assigning roles to roles, i.e., role-a -> assignee -> role-b.
+	delete(attributes, "roles")
+	attributes["role_id"] = schema.StringAttribute{
+		Description: "The ID of the role for access management. If this is changed the resource will be deleted and a new resource will be created.",
```

Comment:
todo: s/ID/UUID/ - ID is vague, UUID let's the user know what the ID is to match the validator.